CodeEdit/Features/CEWorkspace/Models/CEWorkspaceFileManager+DirectoryEvents.swift

```diff
@@ -0,0 +1,206 @@
+//
+//  CEWorkspaceFileManager+DirectoryEvents.swift
+//  CodeEdit
+//
+//  Created by Axel Martinez on 5/8/24.
+//
+
+import Foundation
+
+/// This extension handles the file system events triggered by changes in the root folder.
+extension CEWorkspaceFileManager {
+    /// Called by `fsEventStream` when an event occurs.
+    ///
+    /// This method may be called on a background thread, but all work done by this function will be queued on the main
+    /// thread.
+    /// - Parameter events: An array of events that occurred.
+    func fileSystemEventReceived(events: [DirectoryEventStream.Event]) {
+        DispatchQueue.main.async {
+            var files: Set<CEWorkspaceFile> = []
+            for event in events {
+                // Event returns file/folder that was changed, but in tree we need to update it's parent
+                let parentUrl = "/" + event.path.split(separator: "/").dropLast().joined(separator: "/")
+                // Find all folders pointing to the parent's file url.
+                let fileItems = self.flattenedFileItems.filter({
+                    $0.value.resolvedURL.path == parentUrl
+                }).map { $0.value }
+
+                switch event.eventType {
+                case .changeInDirectory, .itemChangedOwner, .itemModified:
+                    // Can be ignored for now, these I think not related to tree changes
+                    continue
+                case .rootChanged:
+                    // TODO: Handle workspace root changing.
```

Comment:
Could you please create an issue for this after we merge it, if one doesn’t exist yet?

Reply:
I've opened an issue here: https://github.com/CodeEditApp/CodeEdit/issues/1880